modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -487,6 +487,122 @@ func (suite *KeeperTestSuite) TestChanUpgradeAck() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
+	var path *ibctesting.Path
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"channel not found",
+			func() {
+				path.EndpointA.ChannelConfig.PortID = ibctesting.InvalidID
+			},
+			types.ErrChannelNotFound,
+		},
+
+		{
+			"channel state is not in TRYUPGRADE or ACKUPGRADE",
+			func() {
+				suite.Require().NoError(path.EndpointA.SetChannelState(types.OPEN))
+			},
+			types.ErrInvalidChannelState,
+		},
+
+		{
+			"channel has in-flight packets",
+			func() {
+				portID := path.EndpointA.ChannelConfig.PortID
+				channelID := path.EndpointA.ChannelID
+				// Set a dummy packet commitment to simulate in-flight packets
+				suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.SetPacketCommitment(suite.chainA.GetContext(), portID, channelID, 1, []byte("hash"))
+			},
+			types.ErrPendingInflightPackets,
+		},
+		{
+			"flush status is not FLUSHCOMPLETE",
+			func() {
+				channel := path.EndpointA.GetChannel()
+				channel.FlushStatus = types.FLUSHING
+				path.EndpointA.SetChannel(channel)
+			},
+			types.ErrInvalidFlushStatus,
+		},
+		{
+			"connection not found",
+			func() {
+				channel := path.EndpointA.GetChannel()
+				channel.ConnectionHops = []string{"connection-100"}
+				path.EndpointA.SetChannel(channel)
+			},
+			connectiontypes.ErrConnectionNotFound,
+		},
+		{
+			"invalid connection state",
+			func() {
+				connectionEnd := path.EndpointA.GetConnection()
+				connectionEnd.State = connectiontypes.UNINITIALIZED
+				path.EndpointA.SetConnection(connectionEnd)
+			},
+			connectiontypes.ErrInvalidConnectionState,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			expPass := tc.expError == nil
```

Comment:
I would move this line to right before `if expPass {...}` so it's obvious what `expPass` is looking for without needing to go find it.

Reply:
good point, locality is supreme